### ISSUE TITLE
[BUG] Unable to delete files/directories perfectly

### ISSUE THREAD

**Issue by c02y** [cboxdoerfer/fsearch#244]
**Describe the bug**
After I uninstall a program, there are some directories/files were generated by the program(cache and configs such as ~/.config/xxx, ~/.cache/xxx, ~/.local/share/xxx, and the other places in my home directory), so I use fsearch to search those files/directories, I select all of them, and use `Shift-Delete` to delete them, but there are two problems:
1. After deleting them, they are still listed in the result (without refreshing the database)
2. Even after I manually refresh the database, usually, some of them if not all are still in the result, meaning fsearch cannot delete all of them or any of them, even though I tried to update the index database for several times, they are still in the result, so I have to go to my terminal and delete those files/directories by myself.
**To Reproduce**
1. search the keyword of your program
2. select all the files/directories generated in my home directory 
3. delete them using `Shift-Delete`
4. some or all of them are still on the list even after I refresh the index database

**Expected behavior**
1. The best result is after I delete them, they should be removed from the list without updating the index database
2. Even if they are still in the result without updating the index database, after updating the index database manually, they should be gone, I mean, they should be actually removed.

**Screenshots**

**Desktop (please complete the following information):**
 - OS: Manjaro-Cinnamon
 - How you installed FSearch [e.g. built manually, PPA, AUR] Manjaro AUR
 - Version [output of `fsearch --version`] fsearch-git 0.1beta4.r535.g91ab042-1
 - Output of `localectl status`
 ```
$ localectl status
System Locale: LANG=en_US.UTF-8
       VC Keymap: us
      X11 Layout: us
```

**Additional context**

**Comment by cboxdoerfer** [comment]
> After deleting them, they are still listed in the result (without refreshing the database)

Yes, that's a current limitation, but this will be fixed once I finish the file system monitoring.

> delete them using Shift-Delete
> some or all of them are still on the list even after I refresh the index database

Thanks, I can confirm that. The issue is that the current code can only delete files or empty directories, once a directory isn't empty it fails. I'll fix that.
